net/dispatcher.go

```diff
@@ -71,9 +71,12 @@ func (d *Dispatcher) SendRequest(endpoint, method string, jsonData json.RawMessa
 	}
 	updateDispatchHeaders(r, response)
 
-	body, err := ioutil.ReadAll(response.Body)
-	r.Body = body
-	if err != nil {
+	body := make([]byte, maxResponseSize)
+	n, err := io.ReadFull(response.Body, body)
+	r.Body = body[:n]
```

Comment:
@danvixent Any special reason why we're slicing here?

Reply:
we need to trim it down to the size that was actually written. if not the rest of body will contain 0s - because maxResponseSize won't always be reached

Reply:
If `maxResponseSize` is not reached, line 75 should error according to the [docs](https://pkg.go.dev/io#ReadFull). Did you notice a different behaviour in your test?

Reply:
Yes, it returns io.EOF if fewer bytes were read, we ignore EOF.

Reply:
Noted, I ignore EOF which would happen if resp body is empty, will also ignore UnExpectedEOF that will occur if len(resp.Body) < len(body) but resp.Body will be drained.

Reply:
@danvixent This doesn't solve the problem. `r.Body` will be empty if you only ignore `UnExpectedEOF`.